File: frontend/src/api/strategies.ts
import { apiClient } from './client'
import type { BacktestResponse, RecommendationsResponse } from '../types/strategy'

export async function getRecommendations(): Promise<RecommendationsResponse> {
  const response = await apiClient.get<RecommendationsResponse>('/strategies/recommendations')
  return response.data
}

export async function getStrategyBacktest(strategyId = 'ai_daily'): Promise<BacktestResponse> {
  const response = await apiClient.get<BacktestResponse>('/strategies/backtest', {
    params: { strategy_id: strategyId },
  })
  return response.data
}

File: backend/app/api/v1/strategies.py
from fastapi import APIRouter

from app.schemas.strategy import BacktestResponse, RecommendationsResponse
from app.services.strategy_service import StrategyService

router = APIRouter()
strategy_service = StrategyService()


@router.get("/recommendations", response_model=RecommendationsResponse)
async def get_recommendations():
    return strategy_service.get_recommendations()


@router.get("/backtest", response_model=BacktestResponse)
async def get_strategy_backtest(strategy_id: str = "ai_daily"):
    return strategy_service.get_backtest(strategy_id)
